storage/storage.go

```diff
@@ -105,41 +106,48 @@ type Client struct {
 func NewClient(ctx context.Context, opts ...option.ClientOption) (*Client, error) {
 	var host, readHost, scheme string
 
+	// In general, it is recommended to use NewService instead of NewClient, since NewService
```

Comment:
I guess the recommended path is to call NewService without WithHTTPClient, which avoids the need for NewClient?

Reply:
Yes, that's right. Maybe we can consider exporting the internal http client from NewService in the future so it can support the direct-access, client reuse scenario found in reader.go.